src/forms/FieldsetTime.jsx

```diff
@@ -0,0 +1,247 @@
+		const stepSize = KEY_INCREMENT[e.keyCode] * stepMultiplier;
+		const field = e.currentTarget.name; // 'hour' or 'minute'
+		const currentTime = this.props.value
+			? LocalTime.parse(this.props.value)
+			: LocalTime.of(0, 0, 0); // empty value is ok - just start incrementing from 00:00
+
+		const newTimeFields = {
+			hour: currentTime.hour(),
+			minute: currentTime.minute(),
+		};
+		newTimeFields[field] = newTimeFields[field] + stepSize;
+		// wrap the hour when minute reaches boundaries
+		if (newTimeFields.minute >= 60) {
+			newTimeFields.hour = newTimeFields.hour + 1;
+		}
+		if (newTimeFields.minute < 0) {
+			newTimeFields.hour = newTimeFields.hour - 1;
+		}
+
+		// trigger change with positive, modulated hour/minute values
+		const newTime = LocalTime.of(
+			(newTimeFields.hour + 24) % 24,
+			(newTimeFields.minute + 60) % 60,
+			0
+		);
+		this.props.onChange(newTime.toString());
+	};
+
+	render() {
+		const { id, error, disabled, name, is24Hr, value } = this.props;
+
+		const classNames = {
+			fauxInput: cx('fauxInput fauxInput--time', {
+				disabled,
+				error,
+			}),
+			meridian: cx(
+				MERIDIAN_INPUT_CLASS,
+				'flush--all border--none field--reset padding--left',
+				{
+					disabled,
+					error,
+				}
+			),
+		};
+		const displayId = id || name;
+
+		const { hour, minute, meridian } = getValueComponents(is24Hr, value);
+		return (
+			<fieldset className={classNames.fauxInput}>
+				<Flex noGutters>
+					<FlexItem shrink>
+						<input
+							type="number"
+							pattern="\d*"
+							id={`${displayId}-hour`}
+							name={HOURS_INPUT_NAME}
+							min={is24Hr ? 0 : 1}
+							max={is24Hr ? 23 : 12}
+							disabled={disabled}
+							className={`field--reset align--center ${HOURS_INPUT_CLASS}`}
+							onMouseUp={this.selectInputText}
+							onChange={this.onHourChange}
+							onKeyDown={this.onNumberKeyDown}
+							value={hour}
+						/>{' '}
+					</FlexItem>
+					<FlexItem shrink className="align--center">
+						{':'}
+					</FlexItem>
+					<FlexItem shrink>
+						<input
+							type="number"
+							pattern="\d*"
+							id={`${displayId}-minute`}
+							name={MINUTES_INPUT_NAME}
+							min={0}
+							max={59}
+							disabled={disabled}
+							className={`field--reset align--center ${MINUTES_INPUT_CLASS}`}
+							onMouseUp={this.selectInputText}
+							onChange={this.onMinuteChange}
+							onKeyDown={this.onNumberKeyDown}
+							value={minute}
+						/>
+					</FlexItem>
+					{!is24Hr && (
+						<FlexItem
+							shrink
+							className="timeInput-meridianContainer display--flex flex--column flex--center"
+						>
+							<Select
+								id={`${displayId}-meridian`}
+								name="mwc-meridian"
+								className={classNames.meridian}
+								disabled={disabled}
+								value={meridian.toString()} // 0 or 1
+								onChange={this.onMeridianChange}
+							>
+								<option value="0">AM</option>
+								<option value="1">PM</option>
```

Comment:
AM and PM won't ever need to be translated?

Reply:
it's from Latin, so I don't think so - at least, the original implementation didn't take translations for it. I want to say that maybe there are kanji characters for am/pm that we might eventually want to support, but I'm not sure - I think we generally show 24Hr time for languages other than English. Either way, out of scope for this update :)